### PR TITLE
Use tocify JS for dynamic TOC [upstream only]

### PR DESCRIPTION
**Pull request by Lennonka**
#### What changes are you introducing?

Making the table of contents expandable and collapsible.

#### Why are you introducing these changes? (Explanation, links to references, issues, etc.)

Improves UX for Foreman Docs.
For long documents, it's more user friendly and we will be able to display level-3 headings.

#### Anything else to add? (Considerations, potential downsides, alternative solutions you have explored, etc.)

- Using the solution from [StackOverflow: How to use tocify with asciidoctor for a dynamic toc?](https://stackoverflow.com/questions/34481638/how-to-use-tocify-with-asciidoctor-for-a-dynamic-toc) because asciidoctor does not support this natively.
- Not sure about licensing when borrowed from StackOverflow.
- It's not perfect, but better than a static TOC. What do you think?

#### Checklists

* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [x] Foreman 3.13/Katello 4.15 (Satellite 6.17)
* [x] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [x] Foreman 3.11/Katello 4.13 (orcharhino 6.11 on EL8 only)
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

**Comment by Lennonka** [comment]
A few tweaks and increased levels of headings.

It's actually quite cool! :sunglasses:

File: guides/common/docinfo.html [hunk 1]
```diff
@@ -1 +1,74 @@
-<!-- docinfo -->
+<!-- Generate a nice TOC -->
+<script src="https://code.jquery.com/jquery-1.11.3.min.js"></script>
+<script src="https://code.jquery.com/ui/1.11.4/jquery-ui.min.js"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.tocify/1.9.0/javascripts/jquery.tocify.min.js"></script>
```
Comment (maximiliankolb) [inline guides/common/docinfo.html:4]: It would be cool if we checkin these scripts. They are ~350kb in total which is a lot, but I'd be ok with that.

**Review by maximiliankolb — COMMENTED** [review]
one suggestion.

File: guides/common/docinfo.html [hunk 2]
```diff
@@ -1 +1,74 @@
-<!-- docinfo -->
+<!-- Generate a nice TOC -->
+<script src="/js/jquery-1.11.3.min.js"></script>
+<script src="/js/jquery-ui.min.js"></script>
+<script src="/js/jquery.tocify.min.js"></script>
```
Comment (Lennonka) [inline guides/common/docinfo.html:4]: This JS path isn't working and I don't know what the correct path is.

Comment (Lennonka) [reply guides/common/docinfo.html:4]: @ekohl Would you be able to help?

Comment (Lennonka) [reply guides/common/docinfo.html:4]: @maximiliankolb If you know who to ask for help, can you, please? :)

Comment (Thorben-D) [reply guides/common/docinfo.html:4]: @Lennonka I tested your change locally via `make serve` and found it to work nicely!
Checking the [pipeline log](https://github.com/theforeman/foreman-documentation/actions/runs/11843279429/job/33003853408?pr=3453#step:2:68), I noticed that the deployment was using commit `398142fa8e6130035e761c59b8e0632867e38455`. This obviously does not have js libraries, which results in errors.
I don't know why this happens, but it should not block this PR.

**Review by Thorben-D — APPROVED** [review]

**Review by maximiliankolb — APPROVED** [review]
I have seen it work locally with Thorben, so I ACK this change. I suggest that we merge this as is and then wait for it to go live. If it fails, we can always revert this PR if necessary.

**Comment by Lennonka** [comment]
Cherry picked:
- f1009e552a..2f1bd8ee98  3.13 -> 3.13
- 128fd4ebfa..4cc01ec60c  3.12 -> 3.12
- e536de6938..d22d3f0de6  3.11 -> 3.11

**Comment by Lennonka** [comment]
Yay, it's working! :tada:

**Comment by apinnick** [comment]
I suppose it's a bit late for me to comment, but I really dislike this feature now that I have had a chance to use it. It makes reviewing reorganized content extremely tedious because I cannot view more than one folder at the same time.

**Pull request merged**